test/ci-cd/scripts/comfyui/run_in_cluster.sh

```diff
@@ -0,0 +1,189 @@
+# shellcheck disable=SC1091
+source /workspace/build.env 2>/dev/null || true
+if [ -n "${ACP_PLATFORM_BASE_DIR:-}" ]; then
+  # shellcheck disable=SC1091
+  source "${ACP_PLATFORM_BASE_DIR}/use-cases/inference-ref-arch/terraform/_shared_config/scripts/set_environment_variables.sh" 2>/dev/null || true
+fi
+
+# --- Vars (preserved defaults) ---
+export POD_NAME="${POD_NAME:-comfyui-client}"
+export ERROR_FILE="${ERROR_FILE:-/workspace/build-failed.lock}"
+export TEST_WORKFLOW_DIR="${TEST_WORKFLOW_DIR:-test/ci-cd/scripts/comfyui/}"
+
+COMFYUI_PORT="${COMFYUI_PORT:-8188}"
+MAX_WAIT_SECONDS="${MAX_WAIT_SECONDS:-180}"
+STEP_ID=${1:-build-ci} # Default STEP_ID to 'build-ci' if not provided
+
+# --- Helpers ---
+step() { echo -e "\n==== [STEP] $* ====\n"; }
+info() { echo "[INFO] $*"; }
+warn() { echo "[WARN] $*" >&2; }
+
+# This function is triggered by the ERR trap or called directly.
+# It logs an error message and always exits 0.
+log_error() {
+  local exit_code=${1:-$?} # Use provided exit code or last command's
+  local message=${2:-"Script error on line ${BASH_LINENO[0]} (exit code: ${exit_code}) executing: ${BASH_COMMAND}"}
+
+  # Ensure we don't log successful exits when called from the ERR trap
+  if [ "${exit_code}" -eq 0 ] && [ "$#" -eq 0 ]; then
+    return
+  fi
+
+  >"${ERROR_FILE}" # Clear the error file
+  echo "- [${STEP_ID}] ${message}" >>"${ERROR_FILE}"
+  exit 0 # Always exit 0
+}
+trap 'log_error' ERR
+
+# This function runs on any script exit.
+# It cleans up the pod and ALWAYS exits 0.
+cleanup_on_exit() {
+  step "Cleanup"
+  info "Deleting pod: ${POD_NAME} (namespace: ${comfyui_kubernetes_namespace})"
+  kubectl delete pod "${POD_NAME}" -n "${comfyui_kubernetes_namespace}" --ignore-not-found=true >/dev/null 2>&1 || true
+
+  step "Final status"
+  if [ -s "${ERROR_FILE}" ]; then
+    warn "Test run had failures. See '${ERROR_FILE}' for details."
+    step "ERROR FILE CONTENTS"
+    cat "${ERROR_FILE}" || true
+  else
+    info "Test run completed successfully."
+  fi
+
+  # ALWAYS exit 0 per requirement
+  info "Exiting with status 0 to ensure build step passes."
+  exit 0
+}
+trap cleanup_on_exit EXIT
+
+# ------------------------------------------------------------
+# Start client pod
+# ------------------------------------------------------------
+step "Create client pod '${POD_NAME}' in '${comfyui_kubernetes_namespace}'"
+export SA_NAME=$(kubectl get serviceaccount -n ${comfyui_kubernetes_namespace} -o custom-columns=NAME:.metadata.name --no-headers | grep -v "^default$")
+kubectl run "${POD_NAME}" \
+  --image=alpine:latest \
+  --restart=Never \
+  -n "${comfyui_kubernetes_namespace}" \
+  --overrides="{ \"spec\": { \"serviceAccountName\": \"${SA_NAME}\" } }" \
+  --command -- sh -c "apk add --no-cache bash curl jq >/dev/null && echo 'Pod is ready. Waiting...' && sleep 3600"
+
+step "Wait for pod to be Ready"
+kubectl wait --for=condition=Ready "pod/${POD_NAME}" -n "${comfyui_kubernetes_namespace}" --timeout=3600s
```

Comment:
Is there a hard requirement for using a pod here? Could you just set up port-forwarding to the ComfyUI pod? Then you wouldn't need to copy files and everything.